src/commands/timelinecommands.cpp

```diff
@@ -1202,6 +1223,25 @@ void RemoveTransitionByTrimOutCommand::undo()
     if (m_clipIndex > 0) {
         LOG_DEBUG() << "trackIndex" << m_trackIndex << "clipIndex" << m_clipIndex << "delta" << m_delta;
         m_model.addTransitionByTrimIn(m_trackIndex, m_clipIndex, m_delta);
+        // Copy properties from old transition to new transition
+        std::unique_ptr<Mlt::ClipInfo> clipInfo = m_model.getClipInfo(m_trackIndex, m_clipIndex);
+        std::unique_ptr<Mlt::Service> oldService(new Mlt::Producer(MLT.profile(), "xml-string",
```

Comment:
Simplify to `Mlt::Producer oldService(MLT.profile(), "xml-string",`...
Why must it use pointers and `Service`?

Reply:
I copied working code from somewhere else. I don't mind changing it. But it gets complicated because "get_producer()" actually returns a Mlt::Service pointer that needs to be freed by the caller. I guess I can do this:

```
Mlt::Service *tmpNewService = newService.producer();
newService = Mlt::Producer(*tmpNewService);
delete tmpNewService;
```

I'll put a patch together.

Reply:
It has to be service because of a bug in Producer that I just pushed to MLT. But this patch should stay with Service so that it works with released versions of MLT.